### PR TITLE
cabana: support display&edit node name

### PR DESCRIPTION
**Pull request by deanlee**
@sshane I use "Node" as the title, what do you think is a better name for it?

### CODE REVIEW

**Comment by sshane** [comment]
Node is fine

**Comment by sshane** [comment]
The signals can have multiple reading nodes. Maybe `Recipient Nodes`

**Comment by deanlee** [comment]
modified the validator to support multiple nodes, should be ok now.

File: tools/cabana/dbc/dbc.cc [hunk 1]
```diff
@@ -78,6 +78,9 @@ QString cabana::Msg::newSignalName() {
 }
 
 void cabana::Msg::update() {
+  if (transmitter.isEmpty()) {
+    transmitter = "XXX";
+  }
```
Comment (sshane) [inline tools/cabana/dbc/dbc.cc:83]: This function doesn't seem to be called when you exit the message edit dialog so you see `()` until you edit a signal

**Review by sshane — APPROVED** [review]

File: tools/cabana/dbc/dbc.cc (hunk 1 above)
Comment (deanlee) [reply tools/cabana/dbc/dbc.cc:83]: my bad. This is only called when parsing the dbc file.

Comment (deanlee) [reply tools/cabana/dbc/dbc.cc]: fixed

File: tools/cabana/commands.h [hunk 2]
```diff
@@ -10,13 +10,14 @@
 
 class EditMsgCommand : public QUndoCommand {
 public:
-  EditMsgCommand(const MessageId &id, const QString &name, int size, const QString &comment, QUndoCommand *parent = nullptr);
+  EditMsgCommand(const MessageId &id, const QString &name, int size, const QString &node,
+                const QString &comment, QUndoCommand *parent = nullptr);
```
Comment (sshane) [inline tools/cabana/commands.h:14]:
```suggestion
                 const QString &comment, QUndoCommand *parent = nullptr);
```

**Comment by sshane** [comment]
thanks!

**Pull request merged**